experimental/rsocket/RSocketErrors.h

```diff
@@ -0,0 +1,133 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+
+namespace rsocket {
+
+/*
+ * Error Codes from
+ * https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md#error-codes
+ */
+class RSocketError : public std::runtime_error {
+ public:
+  explicit RSocketError(const std::string& s)
+      : std::runtime_error(std::move(s)) {}
+  explicit RSocketError(const char* s) : std::runtime_error(std::move(s)) {}
+  /**
+   * Get the error code for inclusion in an RSocket ERROR frame as per
+   * https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md#error-codes
+   * @return
+   */
+  virtual int getErrorCode() = 0;
+  /**
+   * Get the string name of the error as per
+   * https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md#error-codes
+   * @return
+   */
+  virtual const std::string getErrorString() = 0;
+};
+
+/**
+ * Error Code: INVALID_SETUP 0x00000001
+ */
+class InvalidSetupError : public RSocketError {
+ public:
+  explicit InvalidSetupError(const std::string& s) : RSocketError(s) {}
+  explicit InvalidSetupError(const char* s) : RSocketError(s) {}
+
+  int getErrorCode() override {
+    return 0x00000001;
+  }
+
+  const std::string getErrorString() override {
+    return std::string("INVALID_SETUP");
```

Comment:
rename the method to what()
simply return "INVALID_SETUP"; the std::string ctor is not explicit, so it will convert the char string to std::string.